lib/rules/no-unsafe.js

```diff
@@ -44,7 +45,8 @@ module.exports = {
      */
     function isUnsafe(method) {
       const unsafeMethods = getUnsafeMethods();
-      return unsafeMethods.indexOf(method) !== -1;
+      const isApplicable = versionUtil.testReactVersion(context, '16.3.0');
```

Comment:
This seems like a check we could do earlier, and prevent even returning any visitors in the first place?

Reply:
Added early termination for the rule.